File: array-string/h-index/h-index.ts
function hIndex(citations: number[]): number {
    const n = citations.length;
    const bucket = new Array<number>(n + 1).fill(0);

    for (const c of citations) {
        bucket[c >= n ? n : c]++;
    }

    let papers = 0;
    for (let h = n; h >= 0; h--) {
        papers += bucket[h];
        if (papers >= h) return h;
    }
    return 0;
};

File: array-string/h-index/h-index.cs
public class Solution
{
    public int HIndex(int[] citations)
    {
        int n = citations.Length;
        var bucket = new int[n + 1];

        foreach (int c in citations)
        {
            if (c >= n) bucket[n]++;
            else bucket[c]++;
        }

        int papers = 0;
        for (int h = n; h >= 0; h--)
        {
            papers += bucket[h];
            if (papers >= h) return h;
        }
        return 0;
    }
}